src/main/java/seedu/duke/flashutils/utils/Parser.java

```diff
@@ -40,90 +41,117 @@ private static CommandType parseCommandType(String input) {
-        case View -> createViewCommand(input);
-        case FlashBang -> createFlashbangCommand(input);
-        case Quit -> createQuitCommand();
-        default -> new InvalidCommand();
+            case Add -> createAddCommand(input);
+            case Delete -> createDeleteCommand(input);
+            case Edit -> createEditCommand(input);
+            case View -> createViewCommand(input);
+            case FlashBang -> createFlashbangCommand(input);
+            case Quit -> createQuitCommand();
+            default -> new InvalidCommand();
         };
     }
 
-    public static Command createAddCommand(String input) {
-        Pattern addPattern = Pattern.compile("--m\\s+(.+?)(?:\\s+--t\\s+(.+))?\\s+--q\\s+(.+?)\\s+--a\\s+(.+)");
-        Matcher matcher = addPattern.matcher(input);
+    /**
+     * Splits the given command string into fields based on the provided regular expression.
+     *
+     * <p>Example usage:</p>
+     * <pre>{@code
+     * String rawText = "event <eventname> /from <startDate> /to <endDate>";
+     * String regex = "event\\s(.+)\\s/from\\s(.+)\\s/to\\s(.+)";
+     * ArrayList<String> fields = extractFields(rawText, regex);
+     * // Resulting fields = {<eventname>, <startDate>, <endDate>}
+     * }</pre>
+     *
+     * @param rawText the whole, unaltered command input from the user
+     * @param regex the regular expression used to capture command arguments
+     * @return a list of fields extracted from the input text
+     */
+    private static ArrayList<String> extractFields(String rawText, String regex) {
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(rawText);
+
+        ArrayList<String> fields = new ArrayList<>();
+
         if (matcher.find()) {
-            String moduleName = matcher.group(1);
-            String topic = matcher.group(2);
-            FlashCardSet module = FlashBook.getInstance().getFlashCardSet(moduleName);
-            String question = matcher.group(3);
-            String answer = matcher.group(4);
-            if (topic == null) {
-                topic = "";
+            for (int i = 1; i <= matcher.groupCount(); i++) {
+                fields.add(matcher.group(i));
             }
-            assert !(module == null || question == null || answer == null);
+        }
+
+        return fields;
+    }
+
+    private static FlashCardSet getModule(String moduleName) {
+        return FlashBook.getInstance().getFlashCardSet(moduleName);
+    }
+
+    public static Command createAddCommand(String input) {
+        String addRegex = "--m\\s+(.+?)(?:\\s+--t\\s+(.+))?\\s+--q\\s+(.+?)\\s+--a\\s+(.+)";
+        ArrayList<String> fields = extractFields(input, addRegex);
+
+        if (fields.size() == 4) {
+            String moduleName = fields.get(0);
+            String topic = fields.get(1);
+            String question = fields.get(2);
+            String answer = fields.get(3);
+            FlashCardSet module = getModule(moduleName);
```

Comment:
I think here we should use the assertion
```suggestion
            FlashCardSet module = getModule(moduleName);
            if (topic == null) {
                topic = "";
            }
            assert !(module == null || question == null || answer == null);
```
This is to make sure that if there is no input for the topic field, it will default to the empty string instead of null so that null isn't input into the txt files in data for storage